python/src/spark_rapids_ml/tree.py

```diff
@@ -322,7 +323,7 @@ def _rf_fit(
                 y = pd.concat(y_list)
             else:
                 # should be list of np.ndarrays here
-                X = _concat_and_free(cast(List[np.ndarray], X_list))
+                X = _concat_and_free(cast(List[cp.ndarray], X_list))
```

Comment:
just curious it's possible `cast(cp.ndarray, X)` after `_concat_and_free` is faster?

Reply:
This is for type hinting.   I don't think it would have a perf impact, but might be wrong.

Reply:
that's interesting. Seems you didn't convert the data from the multi-columns scenario into cupy array. https://github.com/NVIDIA/spark-rapids-ml/pull/264/files#diff-80fa68002b29f464f34199afe465010396f9cadce8486f273c3b6eff7cdb3f89R445

Reply:
True.  Haven't tried the multi-columns case, but it might be good to copy to gpu here as well.    Do you have a case where multi-column is faster than single column?